### DISCUSSION TITLE
How to correctly implement pagination for React-Native Flatlist, sometimes rtk query calls two requests when i need only increment the current page to next from redux?

### DISCUSSION THREAD

**Discussion by VladyslavMartynov** [reduxjs/redux-toolkit#2522] [Q&A]
![flatlist](https://user-images.githubusercontent.com/80722986/179421771-45fcad07-ac2d-4c30-ac32-1f590ed7f557.png)

![query](https://user-images.githubusercontent.com/80722986/179421777-f8afc1db-50e9-4db4-8830-abe9c734e341.png)

![redux](https://user-images.githubusercontent.com/80722986/179421778-67b2f917-3f70-4b67-9eb4-77cd7e268ac5.png)

**Comment by phryneas** [comment]
How do you determine that it fetches twice? Just seeing the action twice would be normal in React strict mode, but the second one should be immediately rejected.

**Reply by VladyslavMartynov** [reply]
In Network page, when i make an initial render, everything is okay, but i`m scrolling till the end of the list and it triggered two requests 
<img width="298" alt="image" src="https://user-images.githubusercontent.com/80722986/179456106-13617852-1b15-4a30-9827-3fc765743e13.png">(initial render),

<img width="250" alt="image" src="https://user-images.githubusercontent.com/80722986/179456678-2c4b897c-be3f-4c3c-98cd-2679eebfeea5.png">

**Comment by phryneas** [comment]
those screenshots have abolutely no information value. please open the network requests and actually read what's there.

**Reply by VladyslavMartynov** [reply]
<img width="960" alt="image" src="https://user-images.githubusercontent.com/80722986/179457972-5186981a-e78c-4481-bc6f-acc95eaa4504.png">(initial render)

<img width="960" alt="image" src="https://user-images.githubusercontent.com/80722986/179458240-915d0759-fd5a-4296-9054-2602576be397.png">(double trigger)

**Reply by VladyslavMartynov** [reply]
As i understood when i wanna to reset the pagination to 1 page, i can`t do it, because rtk query remembers that the http request has already existed in network with such params, so it took it from cache and i need to call refetch function from hook in order to make it works with 1 page. But in infinite scrolling something goes wrong as i incrementing the page parametr, it calls two times but i scrolled only once. In redux devtools, i set a page by using action, then it`s normally the previous subscription is removing, execute query, put new data, and then as you can see it triggered setPage again

**Reply by VladyslavMartynov** [reply]
> those screenshots have abolutely no information value. please open the network requests and actually read what's there.

This issue is definetly connected with flatlist bug not with RTK-query, so i found a problem!
